cmake/WoofSettings.cmake

```diff
@@ -29,6 +29,11 @@ function(_checked_add_link_option FLAG)
     endif()
 endfunction()
 
+if(MSVC)
+    # Disable function-level linking
+    _checked_add_compile_option(/Gy-)
+endif()
+
```

Comment:
Adding this compiler flag robustly fixes the latency gremlins I've been seeing (see discussion in #1712)

Reply:
Do you mind if I push a commit to your branch so that you can test the results? Namely reverting [this "fix"](https://github.com/fabiangreffrath/woof/pull/1712#issuecomment-2138096866) to test the theory here.

Reply:
I should have mentioned that I did test that earlier and it does fix the problem, no #1712 required. Also fixes the input_enable weirdness.

Reply:
Interesting, thanks for going through that exercise. I'll let the others comment on the actual fix you're proposing.

Reply:
> Adding this compiler flag robustly fixes the latency gremlins I've been seeing (see discussion in #1712)

Is this for a specific version of MSVC?

Reply:
It's a pretty esoteric option, I'm not sure I understand what it does: [docs](https://learn.microsoft.com/en-us/cpp/build/reference/gy-enable-function-level-linking?view=msvc-170)

Could you please test this artifact: [MSYS2-UCRT64](https://github.com/fabiangreffrath/woof/actions/runs/9428491577/artifacts/1581735331) It is compiled using GCC with a different set of options.

Reply:
> Is this for a specific version of MSVC?

No, it's been around forever as far as I know. I have VS 2019 locally but I've replicated the problem (and the fix) on the CI build artifacts as well.

Reply:
> Could you please test this artifact: [MSYS2-UCRT64](https://github.com/fabiangreffrath/woof/actions/runs/9428491577/artifacts/1581735331) It is compiled using GCC with a different set of options.

Seems fine.

Reply:
You are using Windows 10, right? Is it updated to the latest version? In our default builds we link CRT statically, so maybe there is some sort of problem. It's hard to believe that `/Gy-` can have a noticeable effect.

Reply:
> You are using Windows 10, right? Is it updated to the latest version?

Yeah

>  In our default builds we link CRT statically, so maybe there is some sort of problem.

Static vs dynamic doesn't seem to matter for this. I have both setups locally. LTO=on ofc

> It's hard to believe that `/Gy-` can have a noticeable effect.

I don't find it that hard to believe. It drastically alters the layout of the binary, different optimizer and linker behavior, etc.

Reply:
> I don't find it that hard to believe. It drastically alters the layout of the binary, different optimizer and linker behavior, etc. `-ffunction-sections` is apparently the gcc equivalent

Maybe, but I don't see any difference in performance on my machines. I would rather use `-O2` than touch those options.

Reply:
From this answer, I don't see why we should disable this: https://stackoverflow.com/questions/1834597/what-is-the-comdat-section-used-for

I found it in ZDoom, commit is from 2008 without much comments: https://github.com/ZDoom/gzdoom/commit/fb50df2c6355c3c7e8ca07b3d625b4d8b47a3549#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR17

Reply:
In that case you should add '-ffunction-sections -Wl,--gc-sections' to gcc/clang builds. Those flags are disabled by default. This change is just moving MSVC behavior into line with the other compilers.

Reply:
I think MSVC/GCC developers have their own reasons for setting these defaults. I have a feeling that your latency issues are an SDL/Windows problem, maybe the Woof changes affect this for some reason, but the true problem is not here.

Reply:
The MSVC defaults are designed for large C++ projects. The COMDAT settings make more sense in that context than they do for us here.